packages/flutter/lib/src/material/ink_sparkle.dart

```diff
@@ -0,0 +1,3667 @@
+/// relative to the [referenceBox].
+///
+/// This effect relies on a shader, and therefore hardware acceleration.
+/// Currently, this is only supported by certain C++ engine platforms. The
+/// platforms that are currently supported are Android, iOS, MacOS, and Windows.
+/// Support for CanvasKit web can be tracked here:
+///  - https://github.com/flutter/flutter/issues/85238
+///
+/// To use this effect, pass an instance of [splashFactory] to the
+/// `splashFactory` parameter of either the Material [ThemeData] or any
+/// component that has a `splashFactory` parameter, such as buttons:
+///  - [ElevatedButton]
+///  - [TextButton]
+///  - [OutlinedButton]
+///
+/// The [controller] argument is typically obtained via
+/// `Material.of(context)`.
+///
+/// If [containedInkWell] is true, then the ripple will be sized to fit
+/// the well rectangle, then clipped to it when drawn. The well
+/// rectangle is the box returned by [rectCallback], if provided, or
+/// otherwise is the bounds of the [referenceBox].
+///
+/// If [containedInkWell] is false, then [rectCallback] should be null.
+/// The ink ripple is clipped only to the edges of the [Material].
+/// This is the default.
+///
+/// When the ripple is removed, [onRemoved] will be called.
+///
+/// {@tool snippet}
+///
+/// For typical use, pass the [InkSparkle.splashFactory] to the `splashFactory`
+/// parameter of a button style or [ThemeData].
+///
+/// ```dart
+/// ElevatedButton(
+///   style: ElevatedButton.styleFrom(splashFactory: InkSparkle.splashFactory),
+///   child: const Text('Sparkle!'),
+///   onPressed: () { },
+/// )
+/// ```
+/// {@end-tool}
+class InkSparkle extends InteractiveInkFeature {
+  /// Begin a sparkly ripple effect, centered at [position] relative to
+  /// [referenceBox].
+  ///
+  /// The [color] defines the color of the splash itself. The sparkles are
+  /// always white.
+  ///
+  /// The [controller] argument is typically obtained via
+  /// `Material.of(context)`.
+  ///
+  /// [textDirection] is used by [customBorder] if it is non-null. This allows
+  /// the [customBorder]'s path to be properly defined if it was the path was
+  /// expressed in terms of "start" and "end" instead of
+  /// "left" and "right".
+  ///
+  /// If [containedInkWell] is true, then the ripple will be sized to fit
+  /// the well rectangle, then clipped to it when drawn. The well
+  /// rectangle is the box returned by [rectCallback], if provided, or
+  /// otherwise is the bounds of the [referenceBox].
+  ///
+  /// If [containedInkWell] is false, then [rectCallback] should be null.
+  /// The ink ripple is clipped only to the edges of the [Material].
+  /// This is the default.
+  ///
+  /// Clipping can happen in 3 different ways:
+  ///  1. If [customBorder] is provided, it is used to determine the path for
+  ///     clipping.
+  ///  2. If [customBorder] is null, and [borderRadius] is provided, the canvas
+  ///     is clipped by an [RRect] created from `_clipCallback` and
```

Comment:
Privates shouldn't be referenced in public docs.

Reply:
fixed this